server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -111,6 +118,142 @@ private Optional<String> validateDateOfBirth(String dob) {
     return Optional.empty();
   }
 
+  private Form<EditTiClientInfoForm> validateFirstNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getFirstName())) {
+      return form.withError(FORM_FIELD_NAME_FIRST_NAME, "First name required");
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateLastNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getLastName())) {
+      return form.withError(FORM_FIELD_NAME_LAST_NAME, "Last name required");
+    }
+    return form;
+  }
+
+  private Boolean hasEmailChanged(String newEmail, AccountModel account) {
+    return !newEmail.equals(account.getEmailAddress());
+  }
+
+  private Form<EditTiClientInfoForm> validatePhoneNumber(Form<EditTiClientInfoForm> form) {
```

Comment:
Validation always happens at the DB level (https://github.com/civiform/civiform/blob/3f76aa6c180c781f6adb308a2fc1d2962b11d0c2/server/app/services/CfJsonDocumentContext.java#L107 ) but these validations are needed to make sure the number they have entered is valid. Say they enter 000-000-0000, though its all digits and has 10 digits, it is not a valid number. It is better to catch these issue early and let our users know of the issues before entering it in the system.

Reply:
To clarify what I meant: It looks like `PhoneQuestion` does some similar validation steps in [`validatePhoneNumber`](https://github.com/civiform/civiform/blob/640bade0009756f5713e57feefab533889c4cf02/server/app/services/applicant/question/PhoneQuestion.java#L44-L74) and I was wondering if we could create a common method (e.g. a `PhoneUtils` class) that could be reused in `PhoneQuestion` and in this `TrustedIntermediaryService`  class so that we don't need to duplicate the validation logic (and make sure we don't forget any validations).

I still think it's worth sharing this validation since getting phone numbers right is important, but I think it'd be better to get this PR submitted as-is first, then maybe follow up with a common method for phone number validation.

Reply:
Will followup with this issue: https://github.com/civiform/civiform/issues/6382